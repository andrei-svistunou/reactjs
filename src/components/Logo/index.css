.main-logo__text {
  color: #fc7e8d;
}