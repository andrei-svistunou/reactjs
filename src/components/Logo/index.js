import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Logo = props => (
  <a>
    <span className="main-logo__text">{props.text}</span>
  </a>
);

Logo.propTypes = {
  text: PropTypes.string
};

Logo.defaultProps = {
  text: 'netflixroulett'
};

export default Logo;
