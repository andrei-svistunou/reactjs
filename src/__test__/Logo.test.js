import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Logo from '../components/Logo';

describe('<Logo />', () => {
  test('should render <Logo /> without props', () => {
    const logo = shallow(<Logo />);

    expect(toJson(logo)).toMatchSnapshot();
  });

  test('should render <Logo /> with props', () => {
    const logo = shallow(<Logo text='test' />);

    expect(logo.find('.main-logo__text').text()).toEqual('test');
    expect(toJson(logo)).toMatchSnapshot();
  });
});
