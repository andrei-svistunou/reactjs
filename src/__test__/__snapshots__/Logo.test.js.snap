// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render <Logo /> with props 1`] = `
<a>
  <span
    className="main-logo__text"
  >
    test
  </span>
</a>
`;

exports[`<Logo /> should render <Logo /> without props 1`] = `
<a>
  <span
    className="main-logo__text"
  >
    netflixroulett
  </span>
</a>
`;
